src/base/bittorrent/peerinfo.cpp

```diff
@@ -273,131 +273,136 @@ qreal PeerInfo::relevance() const
     }
 
     // ? = Your client unchoked the peer but the peer is not interested
     if (!isChocked() && !isRemoteInterested()) {
         m_flags += "? ";
-        m_flagsDescription += "? = ";
-        m_flagsDescription += tr("not interested(peer) and unchoked(local)");
-        m_flagsDescription += '\n';
+        flagsDescriptionList.append('?'
+                    + nonBreakingSpacesEquality
+                    + tr("not interested(peer) and unchoked(local)"));
     }
 
     // X = Peer was included in peerlists obtained through Peer Exchange (PEX)
     if (fromPeX()) {
         m_flags += "X ";
-        m_flagsDescription += "X = ";
-        m_flagsDescription += tr("peer from PEX");
-        m_flagsDescription += '\n';
+        flagsDescriptionList.append('X'
+                    + nonBreakingSpacesEquality
+                    + tr("peer from PEX"));
     }
 
     // H = Peer was obtained through DHT
     if (fromDHT()) {
         m_flags += "H ";
-        m_flagsDescription += "H = ";
-        m_flagsDescription += tr("peer from DHT");
-        m_flagsDescription += '\n';
+        flagsDescriptionList.append('H'
+                    + nonBreakingSpacesEquality
+                    + tr("peer from DHT"));
     }
 
     // E = Peer is using Protocol Encryption (all traffic)
     if (isRC4Encrypted()) {
         m_flags += "E ";
-        m_flagsDescription += "E = ";
-        m_flagsDescription += tr("encrypted traffic");
-        m_flagsDescription += '\n';
+        flagsDescriptionList.append('E'
+                    + nonBreakingSpacesEquality
+                    + tr("encrypted traffic"));
     }
 
     // e = Peer is using Protocol Encryption (handshake)
     if (isPlaintextEncrypted()) {
         m_flags += "e ";
-        m_flagsDescription += "e = ";
-        m_flagsDescription += tr("encrypted handshake");
-        m_flagsDescription += '\n';
+        flagsDescriptionList.append('e'
+                    + nonBreakingSpacesEquality
+                    + tr("encrypted handshake"));
     }
 
     // P = Peer is using uTorrent uTP
     if (useUTPSocket()) {
         m_flags += "P ";
-        m_flagsDescription += "P = ";
-        m_flagsDescription += QString::fromUtf8(C_UTP);
-        m_flagsDescription += '\n';
+        flagsDescriptionList.append('P'
+                    + nonBreakingSpacesEquality
+                    + QString::fromUtf8(C_UTP));
     }
 
     // L = Peer is local
     if (fromLSD()) {
         m_flags += "L";
-        m_flagsDescription += "L = ";
-        m_flagsDescription += tr("peer from LSD");
+        flagsDescriptionList.append('L'
+                    + nonBreakingSpacesEquality
+                    + tr("peer from LSD"));
     }
 
     m_flags = m_flags.trimmed();
-    m_flagsDescription = m_flagsDescription.trimmed();
-    if (m_flagsDescription.endsWith('\n', Qt::CaseInsensitive))
-        m_flagsDescription.chop(1);
+    m_flagsDescription = flagsDescriptionList.join('\n').trimmed();
```

Comment:
Can you check if `trimmed()` can be omitted? for both `m_flags` & `m_flagsDescription`, I think its redundant.

Reply:
How?

Reply:
>How?

I'm not asking a formal proof of it, just check whether the strings are equivalent before & after `trimmed()`.

Reply:
Here is indeed not needed but `m_flags` need it.